docs/docs/starlark-reference/directory.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Directory
+sidebar_label: Directory
+---
+
+The `Directory` constructor creates a `Directory` object that represents a directory inside an existing service (see
+the [ServiceConfig][service-config] object).
+Directory object can be either a files artifact or a persistent directory, depending on the arguments passed to the 
+constructor.
+
+```python
+file_artifact_directory = Directory(
+    artifact_name=files_artifact_1,
+)
+# Or:
+persistent_directory = Directory(
+    persistent_key="data-directory"
+)
+```
+
+A directory composed of a files artifact will be automatically provisioned with the given files artifact content and 
```

Comment:
I thought it wasn't either-or (which is why we have them both as possible flags)?

Reply:
It is a XOR right now. I think you need to clarify what the workflow would be when a user specify both a files artifact name and a persistent directory. It's a bit unclear to me